### PR TITLE
fix golangci.yml and some warnings

### PR DESCRIPTION
**Pull request by Vitomir2**
# Description

- update go version to 1.21 and use `go mod tidy`;
- fix `golangci.yml` and some warnings related to property changes;
- add other paths for exclusion rules and make the line length 120 and add the output configuration;
- fix and add the lint, ci, build, and test workflows;
- fetch-depth is used to not doing a shallow checkout which avoids diff error for the linked revision;
- remove the sonar token to avoid the usage of the sonar's analyses;
- remove the race and shuffle flags from the make test command;
- optimize the `TestThrottling` test;
- set a separate `statusCh` variable to store the peer client and remove the `t.Skip()` from the `TestStatusPubSub` test;
- optimize the codebase based on the golintci issues;
- change `DefaultChainID` to `8844`;
- change error string to start with small letter in order to follow the golang conventions;
- actions/setup-go must be done after actions/setup-checkout for cache optimizations.
- updated the version of the actions;
- add submodules: recursive in the checkout action;

# Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [ ] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [x] I have updated the official documentation
- [x] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [x] I have tested this code manually

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -3,14 +3,6 @@
 run:
   timeout: 3m
   tests: true
-  # default is true. Enables skipping of directories:
-  #   vendor$, third_party$, testdata$, examples$, Godeps$, builtin$
-  skip-dirs-use-default: true
-  skip-dirs:
-    - tests
```
Comment (R-Santev) [inline .golangci.yml:10]: why the skip flags are removed

File: consensus/polybft/blockchain_wrapper.go [hunk 2]
```diff
@@ -224,7 +224,12 @@ func (p *blockchainWrapper) UnubscribeEvents(subscription blockchain.Subscriptio
 }
 
 func (p *blockchainWrapper) GetChainID() uint64 {
-	return uint64(p.blockchain.Config().ChainID)
+	chainID := p.blockchain.Config().ChainID
+	if chainID < 0 {
```
Comment (R-Santev) [inline consensus/polybft/blockchain_wrapper.go:228]: better exclude the rule - chainId is always > 0

File: consensus/polybft/polybft.go [hunk 3]
```diff
@@ -30,9 +30,10 @@ import (
 )
 
 const (
-	minSyncPeers = 2
-	pbftProto    = "/pbft/0.2"
-	bridgeProto  = "/bridge/0.2"
+	minSyncPeers   = 2
+	pbftProto      = "/pbft/0.2"
+	bridgeProto    = "/bridge/0.2"
+	DefaultChainID = 8844
```
Comment (R-Santev) [inline consensus/polybft/polybft.go:36]: the old place for this constant is better because chain ID is not polybft-specific term

**Review by R-Santev — CHANGES_REQUESTED** [review]

File: .golangci.yml (hunk 1 above)
Comment (Vitomir2) [reply .golangci.yml:10]: @R-Santev, the `Golangci-lint` warned me that these flags are not allowed anymore in the yml configuration. Therefore:

1. skip-dirs-use-default is always true by default
2. skip-dirs is used as:
```
exclude-dirs:
    - tests
```
3. the service property is also not allowed and there was no difference running it w/o this property.

File: consensus/polybft/blockchain_wrapper.go (hunk 2 above)
Comment (Vitomir2) [reply consensus/polybft/blockchain_wrapper.go:228]: Then, I will have to use the `//no-lint` tag. Is that okay? The linter warns us that there is a possibility of error in the conversion.

**Review by R-Santev — APPROVED** [review]

**Pull request merged**